src/Transpiler.ts

```diff
@@ -335,12 +340,31 @@ export class LuaTranspiler {
         return result + this.indent + "end\n";
     }
 
+    public transpileLoopBody(
+        node: ts.WhileStatement
+            | ts.DoStatement
+            | ts.ForStatement
+            | ts.ForOfStatement
+            | ts.ForInStatement
+    ): string {
+        this.genVarCounter++;
+        this.loopStack.push(this.genVarCounter);
+        let result = this.indent + "do\n";
+        this.pushIndent();
+        result += this.transpileStatement(node.statement);
+        this.popIndent();
+        result += this.indent + "end\n";
+        result += this.indent + `::__continue${this.loopStack[this.loopStack.length - 1]}::\n`;
+        this.loopStack.pop();
+        return result;
+    }
+
     public transpileWhile(node: ts.WhileStatement): string {
         const condition = this.transpileExpression(node.expression);
 
         let result = this.indent + `while ${condition} do\n`;
         this.pushIndent();
```

Comment:
you can push the indent inside the transpileLoopBody function, this will reduce code reuse even more.

Reply:
If you look at the For loop it still transpiles the incrementor within the pushed indent, you couldn't do that if you move the indent manipulation inside. You could technically do only the push but I think that's bad form, the current implementation is probably the nicest way IMO.